packages/components/inputs/date-input/src/date-input.js

```diff
@@ -234,7 +234,6 @@ const DateInput = (props) => {
                   },
                   // we only do this for readOnly because the input
                   // doesn't ignore these events, unlike when its disabled
-                  onFocus: props.isReadOnly ? undefined : openMenu,
```

Comment:
Here is the only line that had to be removed (per each date input) to change the behaviour.